nix/module.nix

```diff
@@ -4,36 +4,45 @@ let
   inherit (lib) mkIf mkOption mkEnableOption;
   cfg = config.services.vast;
   format = pkgs.formats.yaml { };
-  configFile = format.generate "vast.yaml" cfg.settings;
+  configFile' = format.generate "vast.yaml" (lib.recursiveUpdate cfg.settings (lib.importJSON toJsonFile));
+  toJsonFile = pkgs.runCommand "toJson.json" { preferLocalBuild = true; } ''
+    ${pkgs.remarshal}/bin/yaml2json  -i ${cfg.configFile} -o $out
+  '';
```

Comment:
I said the reason at beginning of the comment. Of course you can fully configure vast with nix, but the yaml should also be an option.(personal option for hacking the yaml)

Reply:
Sorry I missed the explanation in the description. I understand now, the solution is quite elegant. I would add an `assert` in case an option is configured in both yaml and nix.

Reply:
We don't need the assert function. you can either keep them both in services or only one. the lib.recursive will be merging or updating all attrs in the final configYAML. Sometimes it is necessary to extract the configuration from the nix to yaml.